src/main/java/net/dv8tion/jda/internal/requests/restaction/interactions/PremiumRequiredCallbackActionImpl.java

```diff
@@ -0,0 +1,42 @@
+package net.dv8tion.jda.internal.requests.restaction.interactions;
+
+import net.dv8tion.jda.api.interactions.callbacks.IPremiumRequiredReplyCallback;
+import net.dv8tion.jda.api.requests.restaction.interactions.InteractionCallbackAction;
+import net.dv8tion.jda.api.requests.restaction.interactions.PremiumRequiredCallbackAction;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.interactions.InteractionImpl;
+import okhttp3.RequestBody;
+
+import javax.annotation.Nonnull;
+import java.util.function.BooleanSupplier;
+
+public class PremiumRequiredCallbackActionImpl extends InteractionCallbackImpl<Void> implements PremiumRequiredCallbackAction
+{
+
+    public PremiumRequiredCallbackActionImpl(IPremiumRequiredReplyCallback interaction)
+    {
+        super((InteractionImpl) interaction);
+    }
+
+    @Override
+    protected RequestBody finalizeData()
+    {
+        return getRequestBody(DataObject.empty()
+                .put("type", InteractionCallbackAction.ResponseType.PREMIUM_REQUIRED.getRaw())
+                .put("data", DataObject.empty()));
```

Comment:
I believe you can set flags on this, like ephemeral

Reply:
It's already ephemeral by default and it's not modifiable I think
> This response will create an ephemeral message shown to the user that ran the interaction, instructing them that whatever they tried to do requires the premium benefits of your app. It also contains an "Upgrade" button to subscribe. The response message is static, but will be automatically updated with the name of your premium SKU.